onflow/flow-go
# [Tool] Export transactions for height range

Close https://github.com/dapperlabs/flow-go/issues/6236


This PR adds a util commandline and admin tool to get transactions by height range to run on execution node or access node.

## How to test
1. start the network, and start the loader to create some txs
```
cd integration/localnet
make init
make start
make load
```
2. run admin tool against execution node 
```
curl localhost:3701/admin/run_command -H 'Content-Type: application/json' -d '{"commandName": "get-transactions", "data": { "start-height": 340, "end-height": 343 }}'
```
This will return all transactions for blocks between height 340 to 343.

3. stop the network, and run util
```
make stop
cd ../../
go run --tags=relic ./cmd/util/main.go export-json-transactions --output-dir ./ --datadir ./integration/localnet/data/access/adbeb15af18a5e4a84d9201cee5205f112d4eda982d5540efba6106b8b02e3d3/protocol --start-height 340 --end-height 343
```
open `transactions.json` to see all transactions grouped by block and collection.
